### ISSUE TITLE
Feature request: case insensitive & magic email

### ISSUE THREAD

**Issue by paolavness** [visualjerk/quill-magic-url#6]
Hi there, noticing that when capitals are used, the url is not picked up. I've added the below to this working on my side to get this working:

```
    magicUrl: {
        globalRegularExpression: /(https?:\/\/\.|mailto:|www.)[\S]+/gi,
        urlRegularExpression: /(https?:\/\/[\S]+)|(mailto:[\S]+)|(www.[\S]+)/i
      }
```

It would also be neat for automagic conversion of email addresses to `mailto:` links. I gave it a stab with the below, which picks up email addresses... But sadly does not magically add the `mailto`.  I'd have to do PR on it and add it to your `normalize()`. Are you open to contributions?

```
    magicUrl: {
        globalRegularExpression: /((https?:\/\/\.|mailto:|www.)[\S]+)|(^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$)/gi,
        urlRegularExpression: /(https?:\/\/[\S]+)|(mailto:[\S]+)|(www.[\S]+)|(^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$)/i
      }
```

**Comment by visualjerk** [comment]
@paolavness Adding mailto for email addresses is a good idea. Contributions are highly appreciated.

**Comment by seanpoulter** [comment]
If you got that regular expression from MDN @paolavness, the last `*` should probably be a `+` or it can match email addresses without a domain like `a@b`.

**Comment by harshmaisheri** [comment]
+1

**Comment by heiCode** [comment]
+1

**Comment by heiCode** [comment]
Hi @visualjerk, I just started to implement the those magic-mails because I need this feature in a project. But in the first step I've opened a [PR](/visualjerk/quill-magic-url/pull/33) to make default regular expressions work case insensitive. Looking forward to receive your Feedback!

**Comment by heiCode** [comment]
Hi @visualjerk, I've created another [PR](https://github.com/visualjerk/quill-magic-url/pull/34) to add detection and conversion of mail addresses. Feel free to merge and/or comment!

I'm already working on a next PR to eleminate the globalExpression options and derive them from the not global ones instaed. What do you think about that? Is there any case a user wanted to use different expression for pasting and typing?

**Comment by heiCode** [comment]
This issue might be closed then, right? The feature is released in Version 3.0.0 (fyi @paolavness, @harshmaisheri)
